src/model-helpers/Document.js

```diff
@@ -0,0 +1,206 @@
+ * Function that generates checksum for uploaded file
+ *
+ * @memberof utils
+ * @function createFileChecksum
+ * @param {fileObjectType} file - Object containing information about file
+ * from form submission (see {@link fileObjectType})
+ * @returns {Promise<WordArray>} Promise - Generates checksum for uploaded file
+ * using the SHA256 algorithm
+ */
+const createFileChecksum = async (file) => {
+  const text = await file.text(); // only hash the first megabyte
+  return sha256(text);
+};
+
+const addAdditionalInfo = (thing, fileObject) => {
+  if (fileObject.type === "Driver's License") {
+    return thing.addUrl(RDF_PREDICATES.about, createDriversLicenseTtlFile(fileObject));
+  }
+  return thing;
+};
+
+export const docDescToThing = async (docDesc, documentUrl, file) => {
+  const checksum = await createFileChecksum(file);
+
+  let thing = buildThing(createThing({ name: docDesc.name }))
+    .addDate(RDF_PREDICATES.uploadDate, new Date())
+    .addStringNoLocale(RDF_PREDICATES.name, docDesc.name)
+    .addStringNoLocale(RDF_PREDICATES.identifier, docDesc.type)
+    .addStringNoLocale(RDF_PREDICATES.sha256, checksum)
+    .addStringNoLocale(RDF_PREDICATES.description, docDesc.description)
+    .addUrl(RDF_PREDICATES.url, `${documentUrl}${file.name}`);
+
+  if (docDesc.date) thing.addDate(RDF_PREDICATES.endDate, new Date(docDesc.date));
+  thing = addAdditionalInfo(thing, docDesc);
+  return thing.build();
+};
+
+export const parseDocument = (documentThing) => {
+  const uploadDate = getDate(documentThing, RDF_PREDICATES.uploadDate);
+  const name = getStringNoLocale(documentThing, RDF_PREDICATES.name);
+  const type = getStringNoLocale(documentThing, RDF_PREDICATES.identifier);
+  const endDate = getDate(documentThing, RDF_PREDICATES.endDate);
+  const checksum = getStringNoLocale(documentThing, RDF_PREDICATES.sha256);
+  const description = getStringNoLocale(documentThing, RDF_PREDICATES.description);
+  const fileUrl = getUrl(documentThing, RDF_PREDICATES.URL);
+  return { uploadDate, name, type, endDate, checksum, description, fileUrl };
+};
+
+export const saveDescription = async (docThing, dataset, session, docUrl) => {
+  const newDataset = setThing(dataset, docThing);
+  const name = getStringNoLocale(docThing, RDF_PREDICATES.name);
+  const doc = await saveSolidDatasetAt(`${docUrl}${name}.ttl`, newDataset, {
+    fetch: session.fetch
+  });
+  return doc;
+};
+
+const createDocumentInternal = async (file, fileDescription, session, docUrl) => {
+  const docThing = await docDescToThing(fileDescription, docUrl, file);
+  const doc = await saveDescription(docThing, createSolidDataset(), session, docUrl);
+  await saveFileInContainer(docUrl, file, {
+    fetch: session.fetch,
+    slug: file.name
+  });
+  return parseDocument(getThingAll(doc)[0]);
+};
+
+export const createDocument = async (file, fileDescription, session, passUrl) => {
+  const { type, name } = fileDescription;
+  const docUrl = `${passUrl}Documents/${type}/${name}/`;
```

Comment:
Think Solid might have a problem with generating a container with the apostrophe in "Driver's License". Think we could just remove it for naming the container.

https://github.com/codeforpdx/PASS/assets/14917816/eded1d50-fbd0-434c-9489-851f5adb1492

Reply:
Fixed